services/QuillLMS/app/models/activity_session.rb

```diff
@@ -376,6 +376,20 @@ def self.delete_activity_sessions_with_no_concept_results(classroom_unit_id, act
     ActivitySession.where(id: incomplete_activity_session_ids).destroy_all
   end
 
+  def self.save_timetracking_data_from_active_activity_session(classroom_unit_id, activity_id)
```

Comment:
As I read it, this function is intended for Lessons, but not other activities.  If that's true, we should probably try to find some way to avoid this being accidentally called on another activity type which I could imagine squashing data for finished sessions that get caught up in the `where` clause below.  I don't think we need to do that protection via code, but a comment here, or more explicit function name would probably solve the problem without doing more.